### PR TITLE
chore: Upgrade CNI to v1.2.0

### PR DESCRIPTION
**Pull request by thatmattlong**
**Reason for Change**:

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
Thanks @thatmattlong!

Does this version of Azure CNI include transparent mode by default in the released tarball?

**Comment by thatmattlong** [comment]
@sozercan  can you please upload the artiifacts to acs mirror?

windows - https://acnreleaseartifacts.azureedge.net/azure-cni/v1.2.0/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.2.0.zip

Linux - https://github.com/Azure/azure-container-networking/releases/download/v1.2.0/azure-vnet-cni-linux-amd64-v1.2.0.tgz

**Comment by thatmattlong** [comment]
Hey @jackfrancis , yes this version is transparent by default

File: pkg/engine/templates_generated.go [hunk 1]
```diff
@@ -11869,12 +11869,8 @@ configureAzureCNI() {
       serviceCidrs={{GetServiceCidr}}
       echo $(cat "$CNI_CONFIG_DIR/10-azure.conflist" | jq  --arg serviceCidrs $serviceCidrs '.plugins[0]+={serviceCidrs: $serviceCidrs}') > /etc/cni/net.d/10-azure.conflist
     fi
-    if [[ "${NETWORK_POLICY}" == "calico" ]]; then
-      sed -i 's#"mode":"bridge"#"mode":"transparent"#g' $CNI_CONFIG_DIR/10-azure.conflist
-    elif [[ "${NETWORK_POLICY}" == "antrea" ]]; then
-      sed -i 's#"mode":"bridge"#"mode":"transparent"#g' $CNI_CONFIG_DIR/10-azure.conflist
-    elif [[ "${NETWORK_POLICY}" == "" || "${NETWORK_POLICY}" == "none" ]] && [[ "${NETWORK_MODE}" == "transparent" ]]; then
-      sed -i 's#"mode":"bridge"#"mode":"transparent"#g' $CNI_CONFIG_DIR/10-azure.conflist
+    if [[ "${NETWORK_MODE}" == "bridge" ]]; then
```
Comment (jackfrancis) [inline pkg/engine/templates_generated.go:11872]: @xuto2 @paulgmiller FYI, I'm inverting the CSE "NetworkMode" logic here to account for the fact that 1.2.0 of Azure CNI is transparent by default in the extracted tarball.

**Comment by jackfrancis** [comment]
> Hey @jackfrancis , yes this version is transparent by default

thanks! I've added some changes to the bootstrap scripts to accommodate that

**Comment by sozercan** [comment]
@thatmattlong artifacts should be uploaded now

**Comment by jackfrancis** [comment]
@sozercan this from ~45 mins ago"

```
==> azure-arm: + timeout 60 curl -fsSL https://kubernetesartifacts.azureedge.net/azure-cni/v1.2.0/binaries/azure-vnet-cni-linux-amd64-v1.2.0.tgz -o /opt/cni/downloads/azure-vnet-cni-linux-amd64-v1.2.0.tgz
==> azure-arm: curl: (22) The requested URL returned error: 404
```

Should I retry?

**Comment by sozercan** [comment]
@jackfrancis if you try again, it might just magically work :) (there was a path issue - fixed now)

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Pull request merged**